### PR TITLE
Stop using the solver for printing sygus synthesis solutions.

### PR DESCRIPTION
**Pull request by abdoo8080**
This PR uses custom methods for printing the synthesis solutions instead of `Solver::printSynthSolution`, which is going to be removed by #6521.

### CODE REVIEW

File: examples/api/utils.cpp [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+std::string defineFunToString(const cvc5::api::Term& f,
+                              const std::vector<cvc5::api::Term> params,
+                              const cvc5::api::Sort& sort,
+                              const cvc5::api::Term body)
+{
+  std::stringstream ss;
+  ss << "(define-fun " << f << " (";
+  std::vector<cvc5::api::Term>::const_iterator it = params.cbegin(),
+                                               cend = params.cend();
+  if (it != cend)
```
Comment (ajreynol) [inline examples/api/utils.cpp:36]: These 4 lines are spurious, if you remove them I think it is equivalent.

**Review by ajreynol — CHANGES_REQUESTED** [review]

File: examples/api/utils.cpp (hunk 1 above)
Comment (abdoo8080) [reply examples/api/utils.cpp:36]: Could you elaborate?

Comment (ajreynol) [reply examples/api/utils.cpp:36]: This is essentially just hard coding one run of the while loop below, if you delete lines 36-40, the code will be the same I believe.

Comment (abdoo8080) [reply examples/api/utils.cpp:36]: They're almost identical. Line 43 prints 1 more space character than line 38.

Comment (abdoo8080) [reply examples/api/utils.cpp]: Replaced with a for loop.

**Review by ajreynol — APPROVED** [review]

**Pull request merged**